modules/market/src/main/java/com/opengamma/strata/market/curve/node/CdsIndexIsdaCreditCurveNode.java

```diff
@@ -0,0 +1,904 @@
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.data.MarketData;
+import com.opengamma.strata.data.ObservableId;
+import com.opengamma.strata.market.curve.IsdaCreditCurveNode;
+import com.opengamma.strata.market.param.DatedParameterMetadata;
+import com.opengamma.strata.market.param.LabelDateParameterMetadata;
+import com.opengamma.strata.market.param.TenorDateParameterMetadata;
+import com.opengamma.strata.product.common.BuySell;
+import com.opengamma.strata.product.credit.Cds;
+import com.opengamma.strata.product.credit.CdsIndex;
+import com.opengamma.strata.product.credit.CdsIndexCalibrationTrade;
+import com.opengamma.strata.product.credit.CdsIndexTrade;
+import com.opengamma.strata.product.credit.CdsQuote;
+import com.opengamma.strata.product.credit.CdsTrade;
+import com.opengamma.strata.product.credit.type.CdsQuoteConvention;
+import com.opengamma.strata.product.credit.type.CdsTemplate;
+import com.opengamma.strata.product.credit.type.DatesCdsTemplate;
+import com.opengamma.strata.product.credit.type.TenorCdsTemplate;
+
+/**
+ * An ISDA compliant curve node whose instrument is a CDS index.
+ * <p>
+ * The trade produced by the node will be a protection payer (BUY) for a positive quantity
+ * and a protection receiver (SELL) for a negative quantity.
+ */
+@BeanDefinition
+public final class CdsIndexIsdaCreditCurveNode
+    implements IsdaCreditCurveNode, ImmutableBean, Serializable {
+
+  /**
+   * The template for the single names associated with this node.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final CdsTemplate template;
+  /**
+   * The label to use for the node.
+   * <p>
+   * When building, this will default based on {@code template} if not specified.
+   */
+  @PropertyDefinition(validate = "notEmpty", overrideGet = true)
+  private final String label;
+  /**
+   * The identifier of the market data value that provides the quoted value.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final ObservableId observableId;
+  /**
+   * The CDS index identifier.
+   * <p>
+   * This identifier is used to refer this CDS index product.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId cdsIndexId;
+  /**
+   * The legal entity identifiers.
+   * <p>
+   * These identifiers refer to the reference legal entities of the CDS index.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ImmutableList<StandardId> legalEntityIds;
+  /**
+   * The market quote convention.
+   * <p>
+   * The CDS index is quoted in par spread, points upfront or quoted spread.
+   * See {@link CdsQuoteConvention} for detail.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final CdsQuoteConvention quoteConvention;
```

Comment:
Should there be a `CdsIndexTemplate` to capture the legal entity IDs, quote convention etc. Lots of code in the `trade()` method below.